src/pages/workspaces/workspace/_tests/WorkspaceMenu.test.js

```diff
@@ -0,0 +1,258 @@
+
+
+jest.mock('src/components/workspace-utils', () => {
+  const originalModule = jest.requireActual('src/components/workspace-utils')
+  return {
+    ...originalModule,
+    useWorkspaceDetails: jest.fn()
+  }
+})
+
+jest.mock('src/components/PopupTrigger', () => {
+  const originalModule = jest.requireActual('src/components/PopupTrigger')
+  return {
+    ...originalModule,
+    MenuTrigger: jest.fn()
+  }
+})
+
+jest.mock('src/components/TooltipTrigger', () => ({
+  ...jest.requireActual('src/components/TooltipTrigger'),
+  __esModule: true,
+  default: jest.fn()
+}))
+
+
+describe('WorkspaceMenu - undefined workspace', () => {
+  beforeEach(() => {
+    useWorkspaceDetails.mockReturnValue({ workspace: undefined })
+    MenuTrigger.mockImplementation(({ content }) => { return div([content]) })
+    TooltipTrigger.mockImplementation(({ content, children }) => {
+      const [open, setOpen] = useState(false)
+      return (div([
+        div(
+          {
+            onMouseEnter: () => {
+              setOpen(true)
+            },
+            onMouseLeave: () => {
+              setOpen(false)
+            }
+          },
+          [children]
+        ),
+        open && !!content && div([content])
+      ]))
+    })
+  })
+
+  it('renders Clone menu item as disabled', () => {
+    const onClone = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onClone },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Clone')
+    expect(menuText).toHaveAttribute('disabled')
+  })
+
+  it('renders Clone tooltip', () => {
+    const onClone = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onClone },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    fireEvent.mouseOver(screen.getByText('Clone'))
+    expect(screen.queryByText('Cloning is not currently supported on Azure Workspaces')).toBeNull()
+  })
+
+  it('renders Share menu item as disabled', () => {
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Share')
+    expect(menuText).toHaveAttribute('disabled')
+  })
+
+  it('renders Share tooltip', () => {
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    fireEvent.mouseOver(screen.getByText('Share'))
+    expect(screen.queryByText('Sharing is not currently supported on Azure Workspaces')).toBeNull()
+  })
+})
+
+
+describe('WorkspaceMenu - GCP workspace', () => {
```

Comment:
I am surprised that `workspaceLoaded` is not getting passed in for any of these tests. I don't understand how they are working because the menu items should always be in the disabled state then.

Reply:
This is because the dynamic path is being executed (call to workspace details).